### ISSUE TITLE
freetype: tests fail

### ISSUE THREAD

**Issue by kamidev** [hexops/mach#575]
Test environment

```
➜  mach git:(main) sw_vers; clang --version
ProductName:	macOS
ProductVersion:	12.6
BuildVersion:	21G115
Apple clang version 14.0.0 (clang-1400.0.29.102)
Target: arm64-apple-darwin21.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
➜  mach git:(main) git log -1 | tee
commit c94dee263bd320ce1629f25e1dad55ebb789cd80
Author: Stephen Gutekanst <stephen@hexops.com>
Date:   Tue Oct 11 01:06:48 2022 +0200

    mach: correct WebGPU C ABI export

    Signed-off-by: Stephen Gutekanst <stephen@hexops.com>
➜  mach git:(main) zig version
0.10.0-dev.4324+c23b3e6fd
```

Failing test
```
➜  mach git:(main) cd libs/freetype
➜  freetype git:(main) zig build test
/Users/jonas/src/zig/mach/libs/freetype/src/Face.zig:76:55: error: expected type '[*c].Users.jonas.src.zig.mach.libs.freetype.zig-cache.o.52778becbd055c66e744ecbe682b71eb.cimport.struct_FT_Open_Args_', found '*const .Users.jonas.src.zig.mach.libs.freetype.zig-cache.o.52778becbd055c66e744ecbe682b71eb.cimport.struct_FT_Open_Args_'
    return intToError(c.FT_Attach_Stream(self.handle, &args.cast()));
                                                      ^~~~~~~~~~~~
/Users/jonas/src/zig/mach/libs/freetype/src/Face.zig:76:55: note: cast discards const qualifier
referenced by:
    attachMemory: /Users/jonas/src/zig/mach/libs/freetype/src/Face.zig:69:16
    refAllDeclsRecursive__anon_8828: /Users/jonas/src/zig-macos-aarch64-0.10.0-dev.4324+c23b3e6fd/lib/std/testing.zig:742:17
    remaining reference traces hidden; use '-freference-trace' to see all reference traces

error: freetype-tests...
error: The following command exited with error code 1:
/Users/jonas/src/zig-macos-aarch64-0.10.0-dev.4324+c23b3e6fd/zig test /Users/jonas/src/zig/mach/libs/freetype/src/main.zig /Users/jonas/src/zig/mach/libs/freetype/zig-cache/o/0cb08e712157b790d4fd164a551126b2/libfreetype.a /Users/jonas/src/zig/mach/libs/freetype/zig-cache/o/4a2c2c5f6370cb28fec8dbc5a0dccff7/libbrotli.a /Users/jonas/src/zig/mach/libs/freetype/zig-cache/o/edc60f2c49348f820f8aa3c8baacc253/libharfbuzz.a -lc -lc -lc++ --cache-dir /Users/jonas/src/zig/mach/libs/freetype/zig-cache --global-cache-dir /Users/jonas/.cache/zig --name freetype-tests --test-no-exec --pkg-begin freetype /Users/jonas/src/zig/mach/libs/freetype/src/main.zig --pkg-end -I /Users/jonas/src/zig/mach/libs/freetype/upstream/freetype/include -I /Users/jonas/src/zig/mach/libs/freetype/upstream/harfbuzz/src --main-pkg-path /Users/jonas/src/zig/mach/libs/freetype --enable-cache
error: the following build command failed with exit code 1:
/Users/jonas/src/zig/mach/libs/freetype/zig-cache/o/0044b2c09f5898f4ffd6132ae038eacf/build /Users/jonas/src/zig-macos-aarch64-0.10.0-dev.4324+c23b3e6fd/zig /Users/jonas/src/zig/mach/libs/freetype /Users/jonas/src/zig/mach/libs/freetype/zig-cache /Users/jonas/.cache/zig test
```

**Comment by alichraghi** [comment]
this has been fixed in https://github.com/hexops/freetype/pull/8, just needs @slimsag approval

**Comment by kamidev** [comment]
Thanks! Then I'll wait for https://github.com/hexops/freetype to be merged. Feel free to close this issue if you want.

**Comment by slimsag** [comment]
Thanks for filing this, much appreciated!

I merged the mentioned PR just now, so I think this should be resolved

**Comment by kamidev** [comment]
Yes, resolved now. 

```
➜  mach git:(main) zig build test
Test [2/138] test.getVersionString...
GLFW version v3.4.0

string: 3.4.0 Cocoa NSGL Null EGL OSMesa monotonic
Test [16/138] test.getRequiredInstanceExtensions... failed to get vulkan instance extensions, error=error.APIUnavailable
Test [48/138] test.setIcon... can't set window icon, wayland maybe? error=error.FeatureUnavailable
Test [92/138] test.setInputModeRawMouseMotion... failed to set input mode, not supported? error=error.PlatformError
All 138 tests passed.
All 6 tests passed.
All 2 tests passed.
All 1 tests passed.
All 9 tests passed.
All 2 tests passed.
Test [2/5] test.connect to device... SKIP
Test [3/5] test.connect to device from descriptor... SKIP
Test [4/5] test.requestDevice behavior: null is_raw... SKIP
Test [5/5] test.requestDevice behavior: invalid id... SKIP
1 passed; 4 skipped; 0 failed.
All 1 tests passed.
➜  mach git:(main) zig version
0.10.0-dev.4333+f5f28e0d2
➜  mach git:(main) git log -1 | tee
commit 78cf48e022c6433a7bf44600e75f9dfb71d70e37
Author: Stephen Gutekanst <stephen@hexops.com>
Date:   Fri Oct 14 11:20:43 2022 -0700

    examples: fix cubemap compilation

    self-hosted compiler picked up alignment issues better.

    fixes hexops/mach#578

    Signed-off-by: Stephen Gutekanst <stephen@hexops.com>
```
